dali/operators/generic/slice/slice_attr.h

```diff
@@ -119,12 +120,20 @@ class SliceAttr {
             }
             slice_end = std::llround(anchor_val + shape_val);
           }
-          DALI_ENFORCE(slice_end <= shape[dim],
-            make_string("Slice end for dim ", dim, " is out of bounds:",
-                        slice_end, ">", shape[dim]));
-          slice.anchor[dim] = std::llround(anchor_val);
+          DALI_ENFORCE(anchor_val >= 0,
+                       make_string("Negative anchor value if forbidden. Got: anchor = ",
+                                   anchor_val));
+          DALI_ENFORCE(slice_end >= 0,
+                       make_string("Negative slice end value if forbidden. Got: slice_end = ",
+                                   slice_end));
+          slice_end = std::min(slice_end, shape[dim]);
+          slice.anchor[dim] = std::min(
+                  ConvertSat<std::remove_reference<decltype(shape)>::type::value_type>(anchor_val),
+                  shape[dim]);
           slice.shape[dim] = slice_end - slice.anchor[dim];
-          assert(slice.anchor[dim] + slice.shape[dim] <= shape[dim]);
+          DALI_ENFORCE(slice.anchor[dim] + slice.shape[dim] <= shape[dim],
```

Comment:
I would stick with assert as this doesn't depend on the user input.

Reply:
Can the `assert` be used anyhow in CI? I wanted this condition to be tested, but I believe we don't test the Debug configuration?

Reply:
No, we are not testing in debug, but if hit that you should get wrong results at the output.